uaclient/contract.py

```diff
@@ -105,6 +106,22 @@ def request_resources(self) -> "Dict[str, Any]":
         )
         return resource_response
 
+    def request_premium_aws_attach(self, pkcs7: str):
```

Comment:
This isn't exactly requesting attach, right? I _think_ this is performing the fetch of the token (which in the normal attach path the user would provide), am I following correctly?

Reply:
Renamed to request_premium_aws_contract_token